spec/features/13_user_adds_profile_photo_spec.rb

```diff
@@ -0,0 +1,45 @@
+require "rails_helper"
+
+feature "profile photo" do
+  let!(:user) { FactoryGirl.create(:user) }
+
+  scenario "new user sees avatar upload option on sign up page" do
+    visit root_path
+    click_link "Sign Up"
+
+    expect(page).to have_content("Add Avatar")
+  end
+
+  scenario "user uploads a profile photo" do
+    visit root_path
+    click_link "Sign Up"
+
+    fill_in "Email", with: user.email
+    fill_in "Username", with: user.username
+    fill_in "First Name", with: user.first_name
+    fill_in "Last Name", with: user.last_name
+    fill_in "Enter Password", with: user.password
+    fill_in "Confirm Password", with: user.password
+    attach_file "Add Avatar", "#{Rails.root}/spec/support/images/twitter-egg-icon.jpg"
+    click_button "Submit"
+    visit user_path(user.id)
+
+    expect(page).to have_content(user.username)
+    expect(page).to have_xpath("//img")
```

Comment:
Your factory sets an image, so this is not actually checking if an image can be uploaded